lib/find.ts

```diff
@@ -2,46 +2,41 @@ import * as fs from 'fs';
 import { join } from 'path';
 import { promisify } from 'util';
 import { MAX_SUPPORTED_FILE_SIZE } from './common';
-import { debug } from './debug';
 
 const readdir = promisify(fs.readdir);
 export const stat = promisify(fs.stat);
 
-export async function find(dir: string): Promise<string[]> {
-  const dirStat = await stat(dir);
-  if (!dirStat.isDirectory()) {
-    throw new Error(`${dir} is not a directory`);
-  }
-
-  const paths = await readdir(dir);
-  const absolutePaths: string[] = [];
-  for (const relativePath of paths) {
-    absolutePaths.push(join(dir, relativePath));
-  }
-
-  return await extractFilePaths(absolutePaths);
+interface FileHandler {
+  (filePath: string, stats: fs.Stats): void;
 }
 
-export async function extractFilePaths(
-  absolutePaths: string[],
-): Promise<string[]> {
+export async function find(src: string): Promise<string[]> {
   const result: string[] = [];
-  for (const absolutePath of absolutePaths) {
-    try {
-      const fileStats = await stat(absolutePath);
-      if (fileStats.isDirectory()) {
-        const subFiles = await find(absolutePath);
-        result.push(...subFiles);
-      } else if (fileStats.isFile()) {
-        if (fileStats.size > MAX_SUPPORTED_FILE_SIZE) {
-          continue;
-        }
-        result.push(absolutePath);
-      }
-    } catch (error) {
-      debug(error.message || `Error reading file ${absolutePath}. ${error}`);
+
+  await traverse(src, (filePath: string, stats: fs.Stats) => {
+    if (!stats.isFile() || stats.size >= MAX_SUPPORTED_FILE_SIZE) {
```

Comment:
the goal was not to skip only files that are greater than 2G?

Reply:
If I recall it right, it fails with current implementation if it's 2GB, so I wanted to skip them as well. That's why I added -1 in common.ts.

To conclude:
1. We should skip 2GB files
2. It should be `>` and not `>=` since I have deducted the const with 1.

Reply:
I've asked as here https://github.com/snyk/snyk-cpp-plugin/pull/48 you raised `fix: skip files greater than 2GB ` and 
```
Due to limitations with nodes Buffer, we agreed
to skip files greater than 2GB.

This is implemented by following changes:

Add file size validation to find.ts
Ticket: TUN-96
```

So if it was not the case we could have the check applied on pr 48 and have the messaging there updated